File: pay/src/main/java/net/yan100/compose/pay/properties/WeChatProperties.java
package net.yan100.compose.pay.properties;


import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

@Data
@ConfigurationProperties(prefix = "compose.pay.wechat")
public class WeChatProperties {

  private String merchantId;
  private String privateKeyPath;
  private String merchantSerialNumber;
  private String certPaths;

  /**
   * appId 小程序 Id
   */
  private String appId;

  /**
   * 异步通知 Id
   */
  private String notifyUrl;

  // api 密钥
  private String appSecret;
  private String appKey;


}

File: psdk/psdk-wxpa/src/main/kotlin/io/github/truenine/composeserver/psdk/wxpa/properties/WechatProperties.kt
package io.github.truenine.composeserver.psdk.wxpa.properties

import io.github.truenine.composeserver.consts.SpringBootConfigurationPropertiesPrefixes
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.NestedConfigurationProperty

@ConfigurationProperties(prefix = SpringBootConfigurationPropertiesPrefixes.PSDK_WXPA)
data class WechatProperties(@NestedConfigurationProperty var wxpa: WxpaProperties = WxpaProperties())
